bert_trainer.py

```diff
@@ -0,0 +1,259 @@
+            text_a=x[DATA_COLUMN],
+            text_b=None,
+            label=x[LABEL_COLUMN]), axis=1)
+
+        test_input_examples = test.apply(lambda x: bert.run_classifier.InputExample(guid=None,
+            text_a=x[DATA_COLUMN],
+            text_b=None,
+            label=x[LABEL_COLUMN]), axis=1)
+
+        def _create_tokenizer_from_hub_module():
+            """Get the vocab file and casing info from the Hub module."""
+            with tf.Graph().as_default():
+                bert_module = hub.Module(self.bert_model_hub)
+                tokenization_info = bert_module(signature="tokenization_info", as_dict=True)
+                with tf.Session() as sess:
+                    vocab_file, do_lower_case = sess.run([tokenization_info["vocab_file"],
+                        tokenization_info["do_lower_case"]])
+
+                    return bert.tokenization.FullTokenizer(
+                            vocab_file=vocab_file, do_lower_case=do_lower_case)
+
+        self.tokenizer = _create_tokenizer_from_hub_module()
+
+        # Convert our train and test features to InputFeatures that BERT understands.
+        if self.is_training:
+            self.train_features = bert.run_classifier.convert_examples_to_features(
+                train_input_examples, self.label_list, self.max_seq_length, self.tokenizer)
+            self.test_features = bert.run_classifier.convert_examples_to_features(
+                test_input_examples, self.label_list, self.max_seq_length, self.tokenizer)
+
+    def _create_model(self, is_predicting, input_ids, input_mask, segment_ids, labels,
+            num_labels):
+        """Creates a classification model."""
+
+        bert_module = hub.Module(
+                self.bert_model_hub,
+                trainable=True)
+        bert_inputs = dict(
+                input_ids=input_ids,
+                input_mask=input_mask,
+                segment_ids=segment_ids)
+        bert_outputs = bert_module(
+                inputs=bert_inputs,
+                signature="tokens",
+                as_dict=True)
+
+        # Use "pooled_output" for classification tasks on an entire sentence.
+        # Use "sequence_outputs" for token-level output.
+        output_layer = bert_outputs["pooled_output"]
+
+        hidden_size = output_layer.shape[-1].value
+
+        # Dense layer that outputs classes after BERT layer.
+        output_weights = tf.get_variable(
+                "output_weights", [num_labels, hidden_size],
+                initializer=tf.truncated_normal_initializer(stddev=0.02))
+
+        print(output_layer.shape, output_weights.shape)
+
+        output_bias = tf.get_variable(
+                "output_bias", [num_labels], initializer=tf.zeros_initializer())
+
+        with tf.variable_scope("loss"):
+
+            # Dropout helps prevent overfitting
+            output_layer = tf.nn.dropout(output_layer, keep_prob=0.9)
```

Comment:
Also I believe usually 0.5 is used for dropout, is there a reason why we use 0.9